pkg/apis/migration.harvesterhci.io/v1beta1/openstack.go

```diff
@@ -44,3 +59,16 @@ func (o *OpenstackSource) GetKind() string {
 func (o *OpenstackSource) GetConnectionInfo() (string, string) {
 	return o.Spec.EndpointAddress, o.Spec.Region
 }
+
+// GetOptions returns the sanitized OpenstackSourceOptions. This means,
+// optional values are set to their default values.
+func (o *OpenstackSource) GetOptions() interface{} {
+	options := o.Spec.OpenstackSourceOptions
+	if options.UploadImageRetryCount == 0 {
+		options.UploadImageRetryCount = OpenstackDefaultRetryCount
+	}
+	if options.UploadImageRetryDelay == 0 {
+		options.UploadImageRetryDelay = OpenstackDefaultRetryDelay
+	}
```

Comment:
Do we want to handle cases when these two values are negative?

Reply:
Done